### DISCUSSION TITLE
ESP32 FreeRTOS.h no such file or directory

### DISCUSSION THREAD

**Discussion by KianYeo** [pschatzmann/arduino-audio-tools#502] [Q&A] [answered]
Hi all,

I am trying to run the streams-url-file example on a ESP32 devkit. However during compilation, its giving an error that FreeRTOS.h is missing.

I have installed the FreeRTOS library from arduino library manager, but this library is incompatible with ESP32. Anyone can advise what I need to do to run this example?

Thanks in advance!

**Comment by pschatzmann** [comment, marked as answer]
FreeRTOS is part of the ESP32 Arduino. So either you did not select the correct board in the board manager or your installation is damaged

**Comment by TangerineDreamer** [comment]
Hi,PHIL,
I got the same error while testing "base-player-a2dp" from the examples-basic-api,i use ESP32 dev module and ESP32 2.02,
so i uploaded your arduino-freertos library,then it works since i got another error:

AudioTools/AudioPrint.h:41:14: error: 'void audio_tools::AudioPrint::flush()' marked 'override', but does not override
         void flush() FLUSH_OVERRIDE  {
              ^~~~~

I think it does not come from freertos but maybe other people encounter such error like me.

**Comment by pschatzmann** [comment]
Please use an actual version: The latest is 2.0.5!
I also just committed a correction which was removing the override...

**Comment by TangerineDreamer** [comment]
Thank you,That's it!!
You've done a very nice work since i was longing for the audiokit sd card to go over a2dp :)
maybe be you can close the old topic i've opened.
